### PR TITLE
Filesystem::joinPaths()

### PR DESCRIPTION
**Pull request by tomasfejfar**
| Q             | A
| ------------- | ---
| Branch?       | 4.4 for features 
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | #26523   
| License       | MIT
| Doc PR        | symfony/symfony-docs#12803

### CODE REVIEW

**Comment by tomasfejfar** [comment]
@javiereguiluz let me know what you think. If it makes sense like this, I'll add docs PR... I also thought about adding another "normalization" function that can normalize separators in any string (`c:\test/app` → `c:/test/app`). That might we useful when one needs to work with paths using string functions for one reason or other... , but I think it's fine to add it in separate PR later.

**Review by darielopper — APPROVED** [review]

File: src/Symfony/Component/Filesystem/Filesystem.php [hunk 1]
```diff
@@ -24,6 +24,27 @@ class Filesystem
 {
     private static $lastError;
 
+    public static function joinPaths(string ...$pathSegments): string
```
Comment (OskarStark) [inline src/Symfony/Component/Filesystem/Filesystem.php]: A description including an example would be helpful

**Review by OskarStark — APPROVED** [review]
Would love to get this merged

File: src/Symfony/Component/Filesystem/Filesystem.php (hunk 1 above)
Comment (tomasfejfar) [reply src/Symfony/Component/Filesystem/Filesystem.php]: Do the docs I've just created help? https://github.com/symfony/symfony-docs/pull/12803

Comment (OskarStark) [reply src/Symfony/Component/Filesystem/Filesystem.php]: Sure thing but the method itself should have a description in the phpdoc too

**Comment by nicolas-grekas** [comment]
I've had a look at existing logic into the wild for similar functions, and I think the currently proposed one is too simplistic: eg golang resolves `..` and `.` references.
There is also PR #30969 with proposed to merge "webmozart/path-util" into the component, which would provide a more advanced implementation of the join() function.
I'm therefore closing, thank you for proposing.

**Pull request closed without merging**